source/building-your-own-javascript-components/index.html.md.erb

```diff
@@ -0,0 +1,282 @@
+  static property = “value”
+}
+```
+
+Otherwise, you can add these properties to your component’s class after its definition:
+
+```js
+class MyComponent extends Component {}
+MyComponent.property = “value”
+```
+
+### Defining the `moduleName` property
+
+Components extending the `Component` class need to define a `moduleName` [static property](#defining-static-properties-on-components), which:
+
+- helps [`createAll` initialise them](#using-createall-with-your-components)
+- supports consistent error messaging
+
+If your project supports native static properties, the `moduleName` property can be defined as a native static property:
+
+```js
+class MyComponent extend Component{
+  static moduleName = “app-my-component”
+}
+```
+
+### Customising initialisation of components
+
+When your component is initialised, it'll need to perform some tasks, such as adding event listeners. You can add these tasks to your component’s `constructor` after calling `super` to set up the base `Component`’s default behaviour.
+
+```js
+class MyComponent extends Component {
+  static moduleName = ‘app-my-component’
+
+  constructor($root) {
+    super($root)
+
+    // Run component specific initialisation here for example
+    this.$root.addEventListener(‘click’, this.handleClick.bind(this))
+  }
+
+  handleClick(event) {
+    // Handle click inside the component’s root
+  }
+}
+```
+
+### Accessing the root element in the component
+
+Within the component’s methods and its constructor, you can access the root element the component was initialised on using `this.$root`.
+
+```js
+class MyComponent extends Component {
+  static moduleName = ‘app-my-component’
+
+  constructor($root) {
+    super($root)
+
+    // Run component specific initialisation here, for example:
+    this.$root.addEventListener(‘click’, this.handleClick.bind(this))
+  }
+
+  handleClick(event) {
+    this.$root.classList.add(‘app-my-component--clicked’)
+  }
+}
+```
+
+#### Customising the expected type of the root element
+
+The `Component` class will verify that the root element is of the right type (by default, `HTMLElement`). You can set an `elementType` [static property](#defining-static-properties-on-components) on your component to define a specific type to check against:
+
+```js
+/**
+ * @extends Component<HTMLAnchorElement> 
+ */
+class LinkEnhancement extends Component {
+  static moduleName = ‘app-link-enhancement’
+
+  // This will make sure that the component can only be initialised
+  // on HTMLAnchorElement
+  static rootElementType = HTMLAnchorElement
+
+  constructor($root) {
+    super($root);
+
+    // this.$root is an `HTMLAnchorElement` and you can access specific properties
+    // like `href` or `hash`
+  }
+}
+```
+
+#### Using `this.$root` with TypeScript
+
+Because `govuk-frontend` does not provide type definitions in its package, TypeScript may give the following error message: “Property ‘$root’ does not exist on type &lt;NAME_OF_YOUR_CLASS&gt;”. 
```

Comment:
I'd just put it as its own paragraph here – this quote is an interesting one, because it's showing a message to *look for*, rather than something we want the service team to tell the user (the usual reason we put things in quotes).

Reply:
> this quote is an interesting one

As in 'we should keep it'?